### PR TITLE
Update packages

### PR DESCRIPTION
**Pull request by jakemac53**
* Updates all packages with `flutter update-packages --force-upgrade`.
* Removes pinning of package:test (issue is resolved)
* Removes pedantic from the package allow list

Pulls in the fix for https://github.com/dart-lang/test/issues/1505

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/update_packages.dart [hunk 1]
```diff
@@ -30,63 +30,63 @@ import '../runner/flutter_command.dart';
 const Map<String, String> kManuallyPinnedDependencies = <String, String>{
   // Add pinned packages here. Please leave a comment explaining why.
   'flutter_gallery_assets': '1.0.2', // Tests depend on the exact version.
-  'flutter_template_images': '4.0.0', // Must always exactly match flutter_tools template.
+  'flutter_template_images':
+      '4.0.0', // Must always exactly match flutter_tools template.
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/update_packages.dart:34]: extraneous newline

Comment (Hixie) [reply packages/flutter_tools/lib/src/commands/update_packages.dart:34]: in fact it looks like a whole bunch of formatting changes were made to this file (it's a pretty good illustration of why we don't use dart format, some of the changes are really much worse than what we have now!).

**Comment by Hixie** [comment]
test-exempt: only rolls dependencies

File: packages/flutter_tools/lib/src/commands/update_packages.dart (hunk 1 above)
Comment (jakemac53) [reply packages/flutter_tools/lib/src/commands/update_packages.dart]: We will have to agree to disagree about the merits of automatic formatting, but I did revert the change :).

**Comment by jakemac53** [comment]
I am seeing this failure https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20flutter_plugins/16597/overview which looks like a timeout, but I don't have any idea how to go about trying to diagnose that?

It appears to be timing out on a `flutter pub get` in the [android_intent](https://github.com/flutter/plugins/blob/master/packages/android_intent/pubspec.yaml) package?

**Comment by jakemac53** [comment]
Ok, it looks like the issue is a version lock due to the `android_intent` package depending on an old version of `build_runner`. I will send a separate PR to update that.

**Comment by jakemac53** [comment]
It looks like this is now failing due to some references to package:pedantic in the analysis options files in some plugins

**Comment by jakemac53** [comment]
cc @Piinks this should be ready for review now

**Comment by jakemac53** [comment]
This should be ready to land now

**Review by jmagman — APPROVED** [review]
LGTM

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Pull request merged**